MessageClickActions: Use dropdowns instead of boolean toggles for better customization of actions

keircn opened the pull request:
1. Fixed the double-encoding issue with reactions
2. Refactored settings to use action dropdowns instead of boolean toggles
3. Added permission checking with toast warnings
4. Added smart context removal (e.g. system messages always disabled)
5. Added configurable click timeout
6. Added toggle for reactions (add/remove)
7. Added toggle for pin/unpin
8. Added quote message feature with selection aware quoting and multi-quote handling (same message = ..., different messages = newlines
9. Added Copy Message ID and Copy User ID actions
10. Removed redundant settings

keircn committed:
fix(MessageClickActions): use reactions constant to construct PUT request for emoji click action

keircn committed:
feat(messageClickActions): toast if we dont have permission to react to the msg

keircn committed:
feat(messageClickActions): explicit action configuration via dropdown selectors
you can now pick actions to fire when we pick up events

keircn committed:
fix(MessageClickActions): call pins correctly and toggle reactions/pins instead of just enabling them and silently failing otherwise

keircn committed:
feat(messageClickActions): add configurable click timeout for triple clicks

keircn committed:
feat(messageClickActions): quoter things

thororen1234 commented:
just right click atp

thororen1234 commented on src/plugins/messageClickActions/index.ts:
```diff
@@ -36,61 +38,225 @@ const focusChanged = () => !WindowStore.isFocused() && (isDeletePressed = false)
+            message: msg,
+            shouldMention: false,
+            showMentionToggle: channel.guild_id !== null
+        });
```
what happened here 😭

keircn committed:
fix(messageClickActions): clean up quoteMessage a little i guess

keircn committed:
refactor(MessageClickActions): make quoteMessage stateless and a little more readable

thororen1234 commented on src/plugins/messageClickActions/index.ts:
```diff
@@ -37,60 +40,194 @@ const focusChanged = () => !WindowStore.isFocused() && (isDeletePressed = false)
+            message: msg,
+            shouldMention: false,
+            showMentionToggle: channel.guild_id !== null
+        });
```
insertTextIntoChatInputBox exists :)

keyage replied:
> insertTextIntoChatInputBox exists :)

I ASKED IF THERE WAS AN EASIER WAY TO DO IT AND NOBODY THOUGHT TO TELL ME THIS????

thororen1234 commented on src/plugins/messageClickActions/index.ts:
```diff
@@ -117,20 +254,41 @@ export default definePlugin({
         const isMe = msg.author.id === myId;
         const isSelfInvokedUserApp = msg.interactionMetadata?.authorizing_integration_owners[ApplicationIntegrationType.USER_INSTALL] === myId;
+        const isDM = !channel.guild_id;
+        const isSystemMessage = !MessageTypeSets.USER_MESSAGE.has(msg.type);
```
`const isSystemMessage = msg.isSystemDM()`

thororen1234 committed:
t

thororen1234 committed:
Merge branch 'dev' into pr/618

thororen1234 committed:
Merge branch 'dev' into pr/618

thororen1234 committed:
use pin actions

keircn committed:
thororen uwahhhhh

imjustprism commented on src/plugins/messageClickActions/index.ts:
```diff
@@ -37,60 +35,197 @@ const focusChanged = () => !WindowStore.isFocused() && (isDeletePressed = false)
+        return;
+    }
+
+    try {
```
useless try catch, pinMessage unpinMessage dont throw sync

imjustprism commented on src/plugins/messageClickActions/index.ts:
```diff
@@ -173,12 +362,35 @@ export default definePlugin({
+            } else if (action === ClickAction.QUOTE) {
+                quoteMessage(channel, msg);
+            } else if (action === ClickAction.REACT) {
+                if (!PermissionStore.can(PermissionsBits.ADD_REACTIONS, channel) && !PermissionStore.can(PermissionsBits.READ_MESSAGE_HISTORY, channel)) {
```
toggleReaction already checks this

imjustprism commented on src/plugins/messageClickActions/index.ts:
```diff
@@ -173,12 +362,35 @@ export default definePlugin({
+                }
+                toggleReaction(channel.id, msg.id, settings.store.reactEmoji, channel, msg);
+            } else if (action === ClickAction.PIN) {
+                if (!PermissionStore.can(PermissionsBits.MANAGE_MESSAGES, channel)) {
```
togglePin already checks this

keircn committed:
refactor(messageClickActions): addresses prisms comments (i cba to write a commit msg)

thororen1234 commented on src/plugins/messageClickActions/index.ts:
```diff
@@ -37,60 +35,197 @@ const focusChanged = () => !WindowStore.isFocused() && (isDeletePressed = false)
+
+    const selection = window.getSelection()?.toString().trim();
+    const content = selection && msg.content?.includes(selection) ? selection : msg.content;
+    if (!content) return;
```
this just needs to be done then I think its good

keyage replied:
no idea but it was probably because of equicord somehow (true)

keyage replied:
no idea but it was probably because of equicord somehow (true)

keircn committed:
feat(messageClickActions): dont use selection in quote action by default
also:
- add reply, edit options to triple click action
- add disable system dms boolean
- fix operator precedence bug in DM check
